src/library/trackset/crate/cratesummary.h

```diff
@@ -1,17 +1,20 @@
 #pragma once
 
+#include <QObject>
+
 #include "library/trackset/crate/crate.h"
 #include "util/duration.h"
 
 // A crate with aggregated track properties (total count + duration)
-class CrateSummary : public Crate {
+class CrateSummary : public QObject, public Crate {
```

Comment:
As already discussed on Zulip it is wrong to *hijack* pure data objects for this purpose.